src/signatures.md

```diff
@@ -167,4 +167,59 @@ let card = BirthdayCard::new("Paul").age(64).text("Happy Valentine's Day!");
 # }
 ```
 
-Note another advantage of builders: Overloaded constructors often don't provide all possible combinations of parameters, whereas with the builder pattern, you can combine exactly the parameters you want.
\ No newline at end of file
+Note another advantage of builders: Overloaded constructors often don't provide all possible combinations of parameters, whereas with the builder pattern, you can combine exactly the parameters you want.
+
+## When must I use `#[must_use]`?
+
+> Use it on Results and mutex locks. - MG
+
+`#[must_use]` causes a compile error if the caller ignores the return value.
+
+Rust functions are often single-purpose. They either:
+
+* Return a value; or
+* Do something (i.e. have side effects) and return nothing.
+
+In neither case do you need to think about `#[must_use]`. (In the first case,
+nobody would call your function unless they were going to use the result.)
+
+`#[must_use]` is useful for those rarer functions which return a result _and_
+have side-effects. In most such cases, it's wise to specify `#[must_use]`, unless
+the return value is truly optional (for example in
+[`HashMap::insert`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.insert)).
+
+## When should I take parameters by value?
+
+Move semantics are more common in Rust than in C++.
+
+> In C++ moves tend to be an optimization, whereas in Rust they're a key semantic part of the program. - MY
+
+To a first approximation, you should assume similar performance when passing
+things by (moved) value or by reference. It's true that a move may turn out to
+be a `memcpy`, but it's often optimized away.
+
+> Express the ownership relationship in the type system, instead of trying to second-guess the compiler for efficiency. - AF
```

Comment:
Suggestion: Move this quote below the paragraph "So here's the heuristic", which I think it's closely related to.